src/core/qgsmaprenderertask.cpp

```diff
@@ -210,14 +229,30 @@ bool QgsMapRendererTask::run()
 
         // build the world file name
         QString outputSuffix = info.suffix();
-        QString worldFileName = info.absolutePath() + '/' + info.baseName() + '.'
-                                + outputSuffix.at( 0 ) + outputSuffix.at( info.suffix().size() - 1 ) + 'w';
-        QFile worldFile( worldFileName );
 
-        if ( worldFile.open( QIODevice::WriteOnly | QIODevice::Truncate ) ) //don't use QIODevice::Text
+        if ( outputSuffix == QStringLiteral( "tif" ) || outputSuffix == QStringLiteral( "tiff" ) )
+        {
+          gdal::dataset_unique_ptr outputDS( GDALOpen( mFileName.toLocal8Bit().constData(), GA_Update ) );
+          if ( outputDS )
+          {
+            double a, b, c, d, e, f;
+            QgsMapSettingsUtils::worldFileParameters( mMapSettings, a, b, c, d, e, f );
+            double geoTransform[] = { c, a, b, f, d, e };
+            GDALSetGeoTransform( outputDS.get(), geoTransform );
+            GDALSetProjection( outputDS.get(), mMapSettings.destinationCrs().toWkt().toLocal8Bit().constData() );
+          }
```

Comment:
else raise an error?

Reply:
Hmm, not sure about the error here. Maybe simply write a word file if the georeferencing embed fails instead?

Reply:
Ok, sounds reasonable